src/Libraries/Microsoft.Extensions.Telemetry/Logging/ExtendedLogger.cs

```diff
@@ -266,6 +265,12 @@ private void ModernPath(LogLevel logLevel, EventId eventId, LoggerMessageState m
             ref readonly MessageLogger loggerInfo = ref loggers[i];
             if (loggerInfo.IsNotFilteredOut(logLevel))
             {
+                if (!config.Sampler.ShouldSample(new SamplingParameters(logLevel, loggerInfo.Category!, eventId)))
```

Comment:
I assume we only want to sample a given record once and then reuse that result for all loggers. It would be quite surprising if a sampling policy of 'Random.Shared.GetNext() > 0.5' meant the message is randomly sent to some providers but not others.

Reply:
But this is a `for` loop through different loggers which can (and usually do) have different categories, and since we pass the `Category` to the samples via `SamplingParameters`, there might be a custom sampler which makes sampling decisions based on the `Category` value.

Reply:
> But this is a for loop through different loggers which can (and usually do) have different categories

Thats not what I'd expect? Unless the R9 implementation is using the same named types to mean something different than they do in the M.E.L implementation, I'd expect you have one ExtendedLogger instance per category. The MessageLoggers array being looped over represents the Logger created by ILoggerProvider.CreateLogger() for that one category.

Reply:
Updated